Build Fails with AlmaLinux / Rocky Linux / CentOS 8 + 9

### Describe the issue

Installing a fresh AlmaLinux 8 (Or Rocky, Or Centos) or Version 9 does not complete building.
Even after installing all the dependencies, doing a 

./zcutil/build.sh --legacy-cpu -j$(nproc)

fails with:
```
[...]
Making all in src
make[1]: Entering directory '/usr/local/src/zen-4.0.0/src'
make[2]: Entering directory '/usr/local/src/zen-4.0.0/src'
make  -C secp256k1 libsecp256k1.la
make[3]: Entering directory '/usr/local/src/zen-4.0.0/src/secp256k1'
make[3]: Leaving directory '/usr/local/src/zen-4.0.0/src/secp256k1'
/bin/sh ../libtool  --tag=CXX   --mode=link /usr/local/src/zen-4.0.0/depends/x86_64-unknown-linux-gnu/share/../native/bin/ccache g++ -m64 -std=c++17  -Wno-nonnull -Wformat -Wformat-security -Wstack-protector -fstack-protector-all  -ffunction-sections -fdata-sections -fPIE -pipe -O1 -g -fopenmp -fwrapv -fno-strict-aliasing -Wno-builtin-declaration-mismatch -Wreturn-type  -pthread   -Wl,-z,relro -Wl,-z,now -pie  -Wl,--gc-sections -L/usr/local/src/zen-4.0.0/depends/x86_64-unknown-linux-gnu/share/../lib  -o zendoo/mcTest zendoo/mcTest-mcTestCallMain.o zendoo_mcTest-base58_codec.o crypto/zendoo_mcTest-sha256.o zendoo_mcTest-uint256.o -lmcTestCall -lzendoo_mc -lpthread -ldl 
libtool: link: /usr/local/src/zen-4.0.0/depends/x86_64-unknown-linux-gnu/share/../native/bin/ccache g++ -m64 -std=c++17 -Wno-nonnull -Wformat -Wformat-security -Wstack-protector -fstack-protector-all -ffunction-sections -fdata-sections -fPIE -pipe -O1 -g -fopenmp -fwrapv -fno-strict-aliasing -Wno-builtin-declaration-mismatch -Wreturn-type -pthread -Wl,-z -Wl,relro -Wl,-z -Wl,now -pie -Wl,--gc-sections -o zendoo/mcTest zendoo/mcTest-mcTestCallMain.o zendoo_mcTest-base58_codec.o crypto/zendoo_mcTest-sha256.o zendoo_mcTest-uint256.o  -L/usr/local/src/zen-4.0.0/depends/x86_64-unknown-linux-gnu/share/../lib -lmcTestCall -lzendoo_mc -lpthread -ldl -pthread -fopenmp
/usr/bin/ld: /usr/local/src/zen-4.0.0/depends/x86_64-unknown-linux-gnu/share/../lib/libmcTestCall.a(mcTestCall.o): relocation R_X86_64_32 against `.rodata' can not be used when making a PIE object; recompile with -fPIE
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:3041: zendoo/mcTest] Error 1
make[2]: Leaving directory '/usr/local/src/zen-4.0.0/src'
make[1]: *** [Makefile:7164: all-recursive] Error 1
make[1]: Leaving directory '/usr/local/src/zen-4.0.0/src'
make: *** [Makefile:642: all-recursive] Error 1
```

oimitting --legacy-cpu does not help either.

```
Flags:

Options used to compile and link:
  with wallet   = yes
  with proton   = no
  with zmq      = yes
  with test     = yes
  debug enabled = no
  wall          = no
  werror        = no

  target os     = linux
  build os      = 

  CC            = /usr/local/src/zen-4.0.0/depends/x86_64-unknown-linux-gnu/share/../native/bin/ccache gcc -m64
  CFLAGS        = -pipe -O1 
  CPPFLAGS      = -I/usr/local/src/zen-4.0.0/depends/x86_64-unknown-linux-gnu/share/../include/  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS -DMULTICORE
  CXX           = /usr/local/src/zen-4.0.0/depends/x86_64-unknown-linux-gnu/share/../native/bin/ccache g++ -m64 -std=c++17
  CXXFLAGS      = -pipe -O1 -g -fopenmp -fwrapv -fno-strict-aliasing -Wno-builtin-declaration-mismatch -Wreturn-type
  LDFLAGS       = -Wl,--gc-sections -L/usr/local/src/zen-4.0.0/depends/x86_64-unknown-linux-gnu/share/../lib 
  ARFLAGS       = cr
```


Thanks for helping :)


### Can you reliably reproduce the issue?

Steps to reproduce in docker:

```
# The basics
FROM almalinux:9
ARG VERSION=4.0.0

# Preparing the System
RUN /usr/bin/dnf groupinstall -y 'Development Tools'
RUN /usr/bin/dnf install -y epel-release
RUN /usr/bin/dnf update -y
RUN /usr/bin/dnf install -y bzip2 git pigz gzip wget pbzip2 perl-interpreter perl-FindBin

# Installing Horizen and preparing
WORKDIR /usr/local/src
RUN wget https://github.com/HorizenOfficial/zen/archive/refs/tags/v${VERSION}.tar.gz -O /usr/local/src/zen-${VERSION}.tar.gz
RUN tar xzf /usr/local/src/zen-${VERSION}.tar.gz
RUN rm -f /usr/local/src/zen-${VERSION}.tar.gz
WORKDIR /usr/local/src/zen-${VERSION}/zcutil

# Building Zen
RUN ./build.sh -j 8
```

The steps can also be done inside a freshly created AlmaLinux 8/9 VM.

### Expected behavior

I have zen binaries.

### Actual behavior and errors

I don't have zen binaries.

### The full version of `zend` you are using

v4.0.0

### Machine specs

Tried different hardware (4 servers and 1 workstation).

### Any extra information that might be useful in the debugging process

_No response_

### Do you have a back-up of `~/.zen` directory and/or a VM snapshot?

_No response_

--- a/depends/packages/libzendoo.mk
+++ b/depends/packages/libzendoo.mk
@@ -6,7 +6,7 @@ $(package)_download_file=$($(package)_git_commit).tar.gz
 $(package)_sha256_hash=e6941f27a1d8612b5a5c61f9d9f3fe0c6c977bb67726819440dc4c4f19f2c733
 $(package)_git_commit=7a1496312b1bcef259aa336e8477e3724a5e0df3
 $(package)_dependencies=rust
-$(package)_patches=cargo.config
+$(package)_patches=cargo.config mcTest-compiler-flag.patch
 
 ifeq ($(host_os),mingw32)
 $(package)_library_file=target/x86_64-pc-windows-gnu/release/libzendoo_mc.a
@@ -35,6 +35,7 @@ $(package)_build_opts_mingw32=--target=x86_64-pc-windows-gnu
 endef
 
 define $(package)_preprocess_cmds
+  patch -p1 < $($(package)_patch_dir)/mcTest-compiler-flag.patch && \
   mkdir -p .cargo && \
   cat $($(package)_patch_dir)/cargo.config | sed 's|CRATE_REGISTRY|$(host_prefix)/$(CRATE_REGISTRY)|' | sed 's|DUMMY_LINKER|$(default_build_CC)|g' > .cargo/config
 endef

--- a/depends/patches/libzendoo/mcTest-compiler-flag.patch
+++ b/depends/patches/libzendoo/mcTest-compiler-flag.patch
@@ -1,0 +1,14 @@
+diff --git a/mc_test/Makefile b/mc_test/Makefile
+index 1d04286..32f6ac5 100644
+--- a/mc_test/Makefile
++++ b/mc_test/Makefile
+@@ -6,7 +6,7 @@ K := $(foreach exec,$(EXECUTABLES),\
+         $(if $(shell PATH="${PATH}" command -v $(exec)),,$(error "No $(exec) in PATH")))
+ 
+ # support also not-so-new compilers
+-CXX_FLAGS += -std=c++11
++CXX_FLAGS += -std=c++11 -fPIE
+ 
+ ifeq ($(shell env | grep RUSTFLAGS),)
+         RUSTFLAGS := -C target-feature=+bmi2,+adx --emit=asm
+

